python/lsst/analysis/drp/colorColorFitPlot.py

```diff
@@ -0,0 +1,385 @@
+        # to get the distances to
+        p1 = np.array([xsFitLine[0], ysFitLine[0]])
+        p2 = np.array([xsFitLine[1], ysFitLine[1]])
+
+        p1HW = np.array([xsFitLine[0], ysFitLineHW[0]])
+        p2HW = np.array([xsFitLine[1], ysFitLineHW[1]])
+
+        distsHW = perpDistance(p1HW, p2HW, zip(xs[fitPoints], ys[fitPoints]))
+        dists = perpDistance(p1, p2, zip(xs[fitPoints], ys[fitPoints]))
+
+        # Now we have the information for the perpendicular line we
+        # can use it to calculate the points at the ends of the
+        # perpendicular lines that intersect at the box edges
+        if fitParams["mHW"] > 5:
+            xMid = (fitParams["yMin"] - fitParams["bODR2"])/fitParams["mODR2"]
+            xs = np.array([xMid - 0.5, xMid, xMid + 0.5])
+            ys = fitParams["mPerp"]*xs + fitParams["bPerpMin"]
+        else:
+            xs = np.array([fitParams["xMin"] - 0.2, fitParams["xMin"], fitParams["xMin"] + 0.2])
+            ys = xs*fitParams["mPerp"] + fitParams["bPerpMin"]
+        ax.plot(xs, ys, "k--", alpha=0.7)
+
+        if fitParams["mHW"] > 5:
+            xMid = (fitParams["yMax"] - fitParams["bODR2"])/fitParams["mODR2"]
+            xs = np.array([xMid - 0.5, xMid, xMid + 0.5])
+            ys = fitParams["mPerp"]*xs + fitParams["bPerpMax"]
+        else:
+            xs = np.array([fitParams["xMax"] - 0.2, fitParams["xMax"], fitParams["xMax"] + 0.2])
+            ys = xs*fitParams["mPerp"] + fitParams["bPerpMax"]
+        ax.plot(xs, ys, "k--", alpha=0.7)
+
+        # Add a histogram
+        axHist.set_ylabel("Number")
+        axHist.set_xlabel("Distance to Line Fit")
+        medDists = np.median(dists)
+        madDists = sigmaMad(dists)
+        meanDists = np.mean(dists)
+
+        rmsDists = np.sqrt(np.mean(np.array(dists)**2))
+        axHist.set_xlim(meanDists - 2.0*rmsDists, meanDists + 2.0*rmsDists)
+        lineMedian = axHist.axvline(medDists, color="k", label="Median: {:0.3f}".format(medDists))
+        lineMad = axHist.axvline(medDists + madDists, color="k", ls="--",
+                                 label="sigma MAD: {:0.3f}".format(madDists))
+        axHist.axvline(medDists - madDists, color="k", ls="--")
+        lineMean = axHist.axvline(meanDists, color="C0", label="Mean: {:0.3f}".format(meanDists))
+        lineRms = axHist.axvline(meanDists + rmsDists, color="C0", ls="--",
+                                 label="RMS: {:0.3f}".format(rmsDists))
+        axHist.axvline(meanDists - rmsDists, color="C0", ls="--")
+
+        linesForLegend = [lineHW, lineInit, lineRefit, fitScatter, lineMedian, lineMad, lineMean, lineRms]
+        fig.legend(handles=linesForLegend, fontsize=8, bbox_to_anchor=(1.0, 0.99),
+                   bbox_transform=fig.transFigure, ncol=2)
+
+        axHist.hist(dists, bins=100, histtype="step", label="Refit", color="C0")
+        axHist.hist(distsHW, bins=100, histtype="step", label="HW", color="C0", alpha=0.5)
```

Comment:
Rather than a fixed number of bins, I generally prefer a fixed bin _width_ so that histograms are easier to interpret. For example, you could say "bins=np.arange(xmin, xmax, binsize)", with binsize=0.01 or something. (If it's too hard, it's probably not vitally important in this case, though...)